### PR TITLE
refactor(run-protocol): vault title vobj

### LINKED ISSUE #4345: Use Virtual Objects in Vaults
## What is the Problem Being Solved?

For MainNet 1, we intend to support durable objects for restart and upgrade.

## Description of the Design

Collect a catalogue of all the persistent data types in Vaults, and convert them to Virtual Objects.

## Security Considerations

This is about durability. It shouldn't have broader security implications.

## Test Plan

Presumably extensive.

### PR DESCRIPTION
**Pull request by turadg**
closes: #4345 

## Description

One last stateful object in vaultFactory.

This is the "outer vault" that end users hold, allowing the "inner vault" that has the assets to be transferred. In the interests of clarity this renames the "outer" entity to "title", since it's a record of ownership of the vault. "Vault" can then be referred to consistently without a qualifier.

In the UI there need not be any distinction. Users will think of a vault and transferring the vault.

### Security Considerations

--

### Documentation Considerations

Outer/inner wasn't documented so nothing to change. Not user facing.

### Testing Considerations

Refactor so existing tests.

### CODE REVIEW

File: packages/run-protocol/src/vaultFactory/vaultTitle.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  getNotifier: ({ state }) => state.notifier,
+  /** @param {MethodContext} context */
+  makeAdjustBalancesInvitation: ({ facets }) =>
+    facets.helper.owned().makeAdjustBalancesInvitation(),
+  /** @param {MethodContext} context */
+  makeCloseInvitation: ({ facets }) =>
+    facets.helper.owned().makeCloseInvitation(),
+  /**
+   * Starting a transfer revokes the outer vault. The associated updater will
+   * get a special notification the the vault is being transferred.
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/vaultTitle.js:63]:
```suggestion
   * Starting a transfer revokes the vaultTitle. The associated updater will
   * get a special notification that the vault is being transferred.
```

File: packages/run-protocol/src/vaultFactory/vaultTitle.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ *   facets: {
+ *     helper: import('@agoric/vat-data/src/types').KindFacet<typeof helper>,
+ *     self: import('@agoric/vat-data/src/types').KindFacet<typeof title>,
+ *   },
+ * }>} MethodContext
+ */
+
+/**
+ *
+ * @param {InnerVault} vault
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/vaultTitle.js:28]: I thought part of the point was that `InnerVault` could become just `Vault`.

File: packages/run-protocol/src/vaultFactory/vaultKit.js [hunk 3]
```diff
@@ -1,62 +1,7 @@
...
-    getCollateralAmount: () => owned(vault).getCollateralAmount(),
-    getCurrentDebt: () => owned(vault).getCurrentDebt(),
-    getNormalizedDebt: () => owned(vault).getNormalizedDebt(),
-  });
-  return { vault, vaultUpdater: updater };
-};
+import { makeVaultTitle } from './vaultTitle';
 
 /**
  * Create a kit of utilities for use of the (inner) vault.
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/vaultKit.js:7]:
```suggestion
 * Create a kit of utilities for use of the vault.
 *
 * @param {Vault} vault
```

**Review by Chris-Hibbert — APPROVED** [review]
Seems fine. I thought part of the point (" 'Vault' can then be referred to consistently without a qualifier.") was to shorten the name of the innerVault.

**Comment by turadg** [comment]
> I thought part of the point (" 'Vault' can then be referred to consistently without a qualifier.") was to shorten the name of the innerVault.

Yes. This enables that. It's separate work from the vobj scope of this PR so I'll do a fast follow.

**Pull request merged**